airflow/contrib/operators/gcp_container_operator.py

```diff
@@ -239,86 +241,35 @@ def __init__(self,
         self.cluster_name = cluster_name
         self.gcp_conn_id = gcp_conn_id
 
+        # Allowing these fields to be templated means that they might be strings.
+        # Make sure to parse them into the correct arg types.
+        if isinstance(impersonation_chain, str):
```

Comment:
The impersonating will be really useful, because we can grant airflow default service account impersonating others then we may no need too many connections in vault,can just use default connection with empty content in vault.